lib/ReactViews/SidePanel/SidePanel.jsx

```diff
@@ -209,6 +212,24 @@ const SidePanel = observer(
   })
 );
 
+// Used to re-route a second re-HOC wrapping of withTerriaRef
+// a better solution is probably to update the HOC to take an array of keys
+const RefForUpload = props => {
+  const SidePanelWrappedForExplore = withTerriaRef(
+    SidePanel,
+    EXPLORE_MAP_DATA_NAME
+  );
+  // it's important we pull outt refFromHOC and not override it again on rest
+  // so that the second HOC can correctly provide the EXPLORE_MAP_DATA_NAME prop
+  const { refFromHOC, ...rest } = props;
+  return (
+    <SidePanelWrappedForExplore refFromHOCForUpload={refFromHOC} {...rest} />
```

Comment:
Neat, but I agree with the above comment. It would be better to allow for more than 1 ref using the original HOC.